prow/tide/github.go

```diff
@@ -416,6 +416,10 @@ func (gi *GitHubProvider) labelsAndAnnotations(instance string, jobLabels, jobAn
 	return
 }
 
+func (gi *GitHubProvider) jobIsRequiredByTide(ps *config.Presubmit, pr *CodeReviewCommon) bool {
+	return ps.ContextRequired() || ps.RunBeforeMerge
```

Comment:
Nit: Add a log to distinguish between these 2 booleans?

Reply:
it's not very common to do so in this codebase. can always add later if there is a need